pkg/controllers/vdb/podfacts.go

```diff
@@ -1204,3 +1204,43 @@ func (p *PodFacts) FindUnsandboxedSubclustersStillInSandboxStatus(scSbInVdbStatu
 	}
 	return sbScMap
 }
+
+// FindSecondarySubclustersWithDifferentImage will scan the secondary subclusters in main cluster
+// and return the secondary subclusters that have different vertica image than main cluster
+func (p *PodFacts) FindSecondarySubclustersWithDifferentImage() []string {
+	scs := []string{}
+	mcImage := ""
+	// find vertica image in primary nodes of main cluster
+	for _, v := range p.Detail {
+		// we expect the pfacts only contains the main cluster pods so we ignore the sandbox pods
+		if v.sandbox != vapi.MainCluster {
+			continue
+		}
+		if v.isPrimary {
+			mcImage = v.image
+			break
+		}
+	}
+	// find secondary subclusters that has a different image
+	seenScs := make(map[string]any)
+	for _, v := range p.Detail {
+		// we expect the pfacts only contains the main cluster pods so we ignore the sandbox pods
+		if v.sandbox != vapi.MainCluster {
```

Comment:
You do not need this as a `PodFacts` object collects the pod facts for a specific sandbox or the main cluster and that info is in `p.SandboxName`

Reply:
We don't know which podFacts the caller will use, if a future caller uses podFacts in a sandbox to call this function, that caller will get an empty slice. That is not likely to happen, but having one more check doesn't hurt.

Reply:
Ok I see. Then update the function description to explain why we need to do this(unsandbox). Also looks like this is only useful for main cluster, then just do an early check at the beginning of the function, something like:
```
if p.GetSandboxName() != vapi.MainCluster {
       return []string{}
}
```